app/products/calls/state/calls_state.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useState} from 'react';
+import {BehaviorSubject} from 'rxjs';
+
+import {CallsState, DefaultCallsState} from '@app/products/calls/types/calls';
+
+const callsStateSubjects = {} as Dictionary<BehaviorSubject<CallsState>>;
+
+const getCallsStateSubject = (serverUrl: string) => {
+    if (!callsStateSubjects[serverUrl]) {
```

Comment:
Wonder if we should do cleanup for this or other parts of the state when a server is removed (i.e. logged out)

The problem may be:
- There is some state on server A
- I remove server A. The state stays until I close the app.
- Without closing the app, add again server A
- I see some artifact (like an ongoing call, for example) because I have a stale state.

Reply:
That's a good observation! But when we reconnect we load the current calls state and set it. Except for the currentCall state, but that would be handled when you end the call.